nucleotide-count: Some cleaning

ferhatelmas opened the pull request:
* use reflect.DeepEqual, instead of custom
 * return error if invalid nucleotide
 * add comments here and there

ferhatelmas commented:
@petertseng @kytrinyx other improvements that are extracted from previous PR.

petertseng commented on exercises/nucleotide-count/nucleotide_count_test.go:
```diff
@@ -69,30 +53,43 @@ func TestCountingDoesntChangeCount(t *testing.T) {
+		if tt.err && err == nil {
+			t.Fatalf("DNA{ %q }: expected error but didn't get one.", tt.strand)
+		} else if !tt.err && err != nil {
+			t.Fatalf("DNA{ %q }: expected no error but got error", tt.strand)
```
here, it will be possible to show what error the student's solution showed (though not strictly necessary)

ferhatelmas replied:
added

petertseng commented:
There are three distinct changes in this one PR, as evidenced by the three bullet points in the description. It would be great if they could be separate commits, since there should be one commit per logical change. This makes the three commits individually easier to understand, instead of having one large commit that is hard to understand (because it contains three unrelated types of changes)

In particular, the "return error if invalid nucleotide" will want a testVersion bump accompanying it (this can go in the same commit as it, since it is essential to it)

petertseng commented:
(Although I forgot to mention this: I agree with these changes and happily welcome them!)

ferhatelmas commented:
@petertseng split it into 3 commits with each logical change. Bumped version and added error message to fatal. Also, rebased. Anything else, please let me know.

petertseng commented:
Thanks, this is all excellent.
